### PR TITLE
Rendering shadows and pipeline fixes

### PR DESCRIPTION
**Pull request by keptsecret**
It works, can be merged  
Ideally, I don't want to yet because likely need cascaded shadows and other small tweaks to camera (exposing frustum values, used later) but there aren't any conflicts so it should be fine.

### CODE REVIEW

**Comment by keptsecret** [comment]
Seems to print "uniform does not exist" but still work just fine

File: src/engine/rendering/Shader.cpp [hunk 1]
```diff
@@ -111,37 +111,31 @@ namespace lei3d
 
 	void Shader::setUniformMat4(const std::string& name, const glm::mat4& matrix) const
 	{
-		bind();
```
Comment (keptsecret) [inline src/engine/rendering/Shader.cpp:114]: @kevinsadi @Bowers-L 
I removed calls to use shader program when setting uniforms, to reduce calls to the OpenGL state for overhead and to help me sift through commands when GPU debugging.
Made all relevant fixes. Pls don't put back.

File: src/engine/rendering/Mesh.hpp [hunk 2]
```diff
@@ -18,6 +18,13 @@ namespace lei3d
 		glm::vec3 Tangent;
 	};
 
+	enum RenderFlag {
```
Comment (kevinsadi) [inline src/engine/rendering/Mesh.hpp:21]: What is happening here?

**Comment by keptsecret** [comment]
Changing this branch into a new features (shadows) and general fixes pr because it needs to be.

**Review by elsien-dev — APPROVED** [review]
I can handle camera changes in my next pr

File: src/game.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#include "core/Application.hpp"
```
Comment (kevinsadi) [inline src/game.cpp:1]: Why is this game.cpp outside of the game folder?

Comment (kevinsadi) [reply src/game.cpp:1]: The same question applies to editor.cpp

**Review by kevinsadi — COMMENTED** [review]
Right now, I'm unable to compile this locally. It may be user error on my part though. I attempted to just checkout the repo and then regenerate the cmake and then run the lei3d project. Notably, I'm getting a `8>c1xx : fatal  error C1083: Cannot open source file: 'F:\Programming\lei3d\src\engine\main.cpp': No such file or directory`. This is on Windows 10, NVIDIA gpu. Visual Studio.

**Review by elsien-dev — CHANGES_REQUESTED** [review]
Make sure building still works before merging.

**Review by elsien-dev — APPROVED** [review]

**Pull request merged**